lib/beaker/hypervisor/docker.rb

```diff
@@ -21,6 +21,19 @@ def initialize(hosts, options)
 
       # Pass on all the logging from docker-api to the beaker logger instance
       ::Docker.logger = @logger
+
+      # Find out what kind of remote instance we are talking against
+      if ::Docker.version['Version'] =~ /swarm/
+        @docker_type = 'swarm'
+        unless ENV['DOCKER_REGISTRY']
+          raise "Using Swarm with beaker requires a private registry. Please setup the private registry and set the 'DOCKER_REGISTRY' env var"
```

Comment:
Could you collect the list of supported docker env vars and put them into the initialize yard doc?